### PR TITLE
Initial multiplayer client

### PR DESCRIPTION
**Pull request by eanders-ms**
Adds a basic multiplayer client app skeleton, with gulp build support.

Please see the `multiplayer/README.md` for more information.

### CODE REVIEW

**Comment by eanders-ms** [comment]
@riknoll The changes I made to the build process look more extensive than they really are. I did some refactoring so that skillmap, authcode, and multiplayer share a build framework. The duplication was getting a bit much. Hopefully it all "just works" as before.

File: multiplayer/src/components/HostGame.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    return (
+        <>
+            <div className="mt-2 h-max">
+                <div className="text-2xl font-bold">{"Hosting a Game"}</div>
+                {netMode === "init" && (
+                    <div className="flex flex-row gap-1 items-end">
+                        <Input
+                            label="Game URL or Share Code"
```
Comment (riknoll) [inline multiplayer/src/components/HostGame.tsx:40]: seems like `lf()` is missing on most the localizable strings in this file

**Review by jwunderl — APPROVED** [review]

File: multiplayer/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "arrowParens": "avoid",
+    "semi": false,
```
Comment (jwunderl) [inline multiplayer/.prettierrc:3]: nit: I am a little curious on this one, I would say most of our codebases uses semicolons / I personally find it easier to read; any particular reason to override prettier default?

Comment (riknoll) [reply multiplayer/.prettierrc:3]: i too prefer semicolons

**Review by riknoll — APPROVED** [review]
LGTM, but missing a lot of localized string in join/host game files

File: multiplayer/.prettierrc (hunk 2 above)
Comment (eanders-ms) [reply multiplayer/.prettierrc:3]: I can go either way, so with-semicolons it is.  I wonder how many terabytes of semicolons exist in github's storage.

**Pull request merged**